ddtrace/profiling/profiler.py

```diff
@@ -209,9 +219,13 @@ def __attrs_post_init__(self):
         exporters = self._build_default_exporters()
 
         if exporters:
-            self._scheduler = scheduler.Scheduler(
-                recorder=r, exporters=exporters, before_flush=self._collectors_snapshot
-            )
+            if self._lambda_function_name:
+                scheduler_class = (
+                    serverless_scheduler.ServerlessScheduler
+                )  # type: Union[type[scheduler.Scheduler], type[serverless_scheduler.ServerlessScheduler]]
```

Comment:
I don't think you need type then here
otherwise `typing.Type[serverless_scheduler.ServerlessScheduler]` or `typing.Type[sscheduler.Scheduler]` should be what you want I guess.

Reply:
See above comment, this fails mypy